web/src/components/fullscreen.tsx

```diff
@@ -0,0 +1,38 @@
+import { createEffect, createSignal, onCleanup } from "solid-js"
+
+export const FullscreenButton = () => {
+	const [isActive, setIsActive] = createSignal(false)
+
+	const toggleFullscreen = () => {
+		const videoElement = document.getElementById("video")
+
+		if (!isActive()) {
+			void videoElement?.requestFullscreen()
+			setIsActive(true)
+		} else {
+			void document.exitFullscreen()
+			setIsActive(false)
+		}
+	}
+	createEffect(() => {
+		const handleFullscreenChange = () => {
+			setIsActive(!!document.fullscreenElement)
+		}
+
+		document.addEventListener("fullscreenchange", handleFullscreenChange)
+		onCleanup(() => document.removeEventListener("fullscreenchange", handleFullscreenChange))
+	})
+
+	return (
+		<button
+			class="
+				flex h-4 w-4 items-center justify-center rounded bg-transparent
+				p-4 text-white hover:bg-black/80 focus:outline-none
+				"
+			onClick={toggleFullscreen}
```

Comment:
It would be cool to add a keyboard for toggling on/off this button, too! Maybe not in scope here, but `f` is generally the key used for going in and out of fullscreen when the player is focused.